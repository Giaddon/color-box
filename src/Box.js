import React from "react";

function Box({width, height, color, remove, id}) {
  const style = {

    backgroundColor: color,
    height: height,
    width: width,
    margin: "20px auto",
  }
  const buttonStyle = {
    position: "relative",
    left: "40px",
  }

  const handleClick = () => {
    remove(id);
  }

  return (
    <div style={style}>
      <button style={buttonStyle} onClick={handleClick}>X</button>
    </div>
      
  )
}

export default Box;