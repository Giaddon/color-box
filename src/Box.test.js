import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Box from './Box';



//smoke test
it("renders without crashing", function () {
  render(<Box />);
});


//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(< Box />);
  expect(asFragment()).toMatchSnapshot();
});


//looks for button on Box to make sure it renders properly
it("has the X button", function () {
  const { asFragment, getByRole } = render(< Box />);
   let button = getByRole("button");
  
expect(button).toBeInTheDocument()

});
