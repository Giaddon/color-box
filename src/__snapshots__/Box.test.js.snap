// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="margin: 20px auto;"
  >
    <button
      style="position: relative; left: 40px;"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
